mysql/check.py

```diff
@@ -1008,24 +1014,53 @@ def _get_stats_from_innodb_status(self, db):
                 results['Innodb_os_file_writes'] = long(row[4])
                 results['Innodb_os_file_fsyncs'] = long(row[8])
             elif line.find('Pending normal aio reads:') == 0:
-                # Pending normal aio reads: 0, aio writes: 0,
-                # or Pending normal aio reads: 0 [0, 0] , aio writes: 0 [0, 0] ,
-                # or Pending normal aio reads: [0, 0, 0, 0] , aio writes: [0, 0, 0, 0] ,
-                # or Pending normal aio reads: 0 [0, 0, 0, 0] , aio writes: 0 [0, 0, 0, 0] ,
-                if len(row) == 14:
-                    results['Innodb_pending_normal_aio_reads'] = long(row[4])
-                    results['Innodb_pending_normal_aio_writes'] = long(row[10])
+                if len(row) == 8:
+                    # (len(row) == 8)  Pending normal aio reads: 0, aio writes: 0,
+                    if row[4].isdigit() and row[7].isdigit():
+                        results['Innodb_pending_normal_aio_reads'] = long(row[4])
+                        results['Innodb_pending_normal_aio_writes'] = long(row[7])
+                    else:
+                        self.log.warning("Can't parse result line %s" % line)
+
+                elif len(row) == 14:
+                    # (len(row) == 14) Pending normal aio reads: 0 [0, 0] , aio writes: 0 [0, 0] ,
+                    if row[4].isdigit() and row[10].isdigit():
+                        results['Innodb_pending_normal_aio_reads'] = long(row[4])
+                        results['Innodb_pending_normal_aio_writes'] = long(row[10])
+                    else:
+                        self.log.warning("Can't parse result line %s" % line)
+
                 elif len(row) == 16:
-                    results['Innodb_pending_normal_aio_reads'] = (long(row[4]) + long(row[5]) +
-                                                                  long(row[6]) + long(row[7]))
-                    results['Innodb_pending_normal_aio_writes'] = (long(row[11]) + long(row[12]) +
-                                                                   long(row[13]) + long(row[14]))
+                    # (len(row) == 16) Pending normal aio reads: [0, 0, 0, 0] , aio writes: [0, 0, 0, 0] ,
+                    if self._are_values_numeric(row[4:8]) and self._are_values_numeric(row[11:15]):
+                        results['Innodb_pending_normal_aio_reads'] = (long(row[4]) + long(row[5]) +
```

Comment:
Wouldn't this fail for row[4]: `long('[0')` for example? I think we have to sanitize that: 
eg. `long(row[4].strip('['))`

Similarly, for row[7]: `long(row[7])` ---> `long(row[7].strip(']'))`